packages/samples/headless-react/src/components/facet/facet.fn.hooks.tsx

```diff
@@ -0,0 +1,55 @@
+import {FunctionComponent} from 'react';
+import {FacetSearch} from './facet-search';
+import {useController} from './facet-hook';
+import {Facet as HeadlessFacet} from '@coveo/headless';
+
+interface FacetProps {
+  controller: HeadlessFacet;
+}
+
+export const FacetWithHook: FunctionComponent<FacetProps> = ({controller}) => {
+  const [state] = useController(controller);
+  return (
+    <ul>
+      <li>
+        <FacetSearch
+          controller={controller.facetSearch}
+          facetState={state.facetSearch}
+          isValueSelected={(facetSearchValue) =>
+            !!state.values.find(
+              (facetValue) =>
+                facetValue.value === facetSearchValue.displayValue &&
+                controller.isValueSelected(facetValue)
+            )
+          }
+        />
+      </li>
+      <li>
+        <ul>
+          {state.values.map((value) => (
+            <li key={value.value}>
+              <input
+                type="checkbox"
+                checked={controller.isValueSelected(value)}
+                onChange={() => controller.toggleSelect(value)}
+                disabled={state.isLoading}
+              />
+              {value.value} ({value.numberOfResults} results)
+            </li>
+          ))}
+        </ul>
+      </li>
+    </ul>
+  );
+};
+
+// usage
+
+/**
+ * ```tsx
+ * const options: FacetOptions = {field: 'objecttype'};
+ * const controller = buildFacet(engine, {options});
+ *
+ * <Facet controller={controller} />;
```

Comment:
I think the React package should abstract the controllers more than this. I would expect the React `Facet` to only need options by default. It would handle getting the engine via provider, initializing the facet, subscribing, and rendering a default view if no children are specified.

```
<SearchEngineProvider>
    <Facet field={...}/>
</SearchEngineProvider>
```

People who want to customize the default view can define their own inside the Facet, and add interactivity using the controller instance.

```
<SearchEngineProvider>
    <Facet field={...}>
        (facet) => {
          return <ul>{facet.state.values.map( ... list items ....)}</ul>
        }
    </Facet>
</SearchEngineProvider>
```

A developer could also choose to place content of the custom facet in a separate file

```
<SearchEngineProvider>
    <Facet field={...}>
        (facet) => <MyFacet controller={facet}/>
    </Facet>
</SearchEngineProvider>
```